### PR TITLE
Bugfix/itdev 35705 allocations uniqueness by status and path

### PR DESCRIPTION
**Pull request by melonistic**
Manually Testing:

 - create an allocation and move it to ready for Deletion. Create another and move to active.
 - run the ingest quotas task.  In main, it will throw an exception.  In this branch, it should be successful.
 - The History should be recorded in the active allocation, but not the "ready for deletion" allocation

### CODE REVIEW

File: coldfront/plugins/qumulo/tests/test_quotas.py [hunk 1]
```diff
@@ -24,91 +27,102 @@
 
 def coldfront_allocations() -> str:
```
Comment (melonistic) [inline coldfront/plugins/qumulo/tests/test_quotas.py:28]: Black formatter changes

File: coldfront/plugins/qumulo/tests/test_quotas.py [hunk 2]
```diff
@@ -24,91 +27,102 @@
...
+        "/storage2/fs1/daryls3/Active/teamfolder": {"limit": "6597069766656"},
+        "/storage2/fs1/kdandurand": {"limit": "5497558138880"},
+        "/storage2/fs1/vtieppofrancio/Active/Human Subjects Research": {
+            "limit": "5497558138880"
+        },
+        "/storage2/fs1/kirilloff": {"limit": "5497558138880"},
     }
 
 
+# fmt: off
```
Comment (melonistic) [inline coldfront/plugins/qumulo/tests/test_quotas.py:125]: Skip formatting to keep the format from QUMULO.

File: coldfront/plugins/qumulo/tests/test_quotas.py [hunk 3]
```diff
@@ -744,3 +777,82 @@ def test_after_getting_daily_usages_from_qumulo_api(
...
+            "storage_quota": limit,
+            "protocols": ["nfs"],
+            "rw_users": [f"user_{index}_rw"],
+            "ro_users": [f"user_{index}_ro"],
+            "storage_ticket": f"ITSD-{index}",
+            "cost_center": "Uncle Pennybags",
+            "department_number": "Time Travel Services",
+            "service_rate": "general",
+        }
+        allocation_active = create_allocation(
```
Comment (melonistic) [inline coldfront/plugins/qumulo/tests/test_quotas.py:819]: The next two allocations have the same path

File: coldfront/plugins/qumulo/tests_integration/utils/test_qumulo_api/test_list_qumulo_quota_usages.py [hunk 4]
```diff
@@ -9,12 +9,13 @@
     print_all_quotas_with_usage,
 )
 
-BLANK = ''
```
Comment (melonistic) [inline coldfront/plugins/qumulo/tests_integration/utils/test_qumulo_api/test_list_qumulo_quota_usages.py:12]: The Black formatter caused all the changes in this file.

**Comment by harterjwustl** [comment]
@melonistic , please provide instructions on how to test

File: coldfront/plugins/qumulo/tests/test_quotas.py [hunk 5]
```diff
@@ -744,3 +777,84 @@ def test_after_getting_daily_usages_from_qumulo_api(
...
+        allocation_pending.status = self.status_pending
+        allocation_pending.save()
+
+        exceptionRaised = False
+        try:
+            ingest_quotas_with_daily_usage()
+        except:
+            exceptionRaised = True
+
+        self.assertFalse(exceptionRaised)
```
Comment (harterjwustl) [inline coldfront/plugins/qumulo/tests/test_quotas.py:837]:
```suggestion
        try:
            ingest_quotas_with_daily_usage()
        except:
            self.fail("ingest_quotas failed")
```

**Review by harterjwustl — APPROVED** [review]
Approving, other than not of suggested change

**Comment by harterjwustl** [comment]
I undid my Approval because I still need instructions for and to do a qa test...but the code looks good

**Comment by melonistic** [comment]
> @melonistic , please provide instructions on how to test

@harterjwustl

`python manage.py test coldfront.plugins.qumulo.tests.test_quotas`

In particular, the (last) test case, `test_filtering_out_not_active_allocations,` shows how the active allocation is selected amongst the others despite all having the same path.

To test manually, create an allocation and move it to ready for Deletion. Create another and move to active.

**Comment by harterjwustl** [comment]
> > @melonistic , please provide instructions on how to test
> 
> @harterjwustl
> 
> `python manage.py test coldfront.plugins.qumulo.tests.test_quotas`
> 
> In particular, the (last) test case, `test_filtering_out_not_active_allocations,` shows how the active allocation is selected amongst the others despite all having the same path.
> 
> To test manually, create an allocation and move it to ready for Deletion. Create another and move to active.

What are the expected results in manual testing?  How will I know it works vs whats in main?

File: coldfront/plugins/qumulo/tests/test_quotas.py [hunk 6]
```diff
@@ -10,8 +10,11 @@
 from coldfront.core.allocation.models import (
     AllocationAttribute,
     AllocationAttributeType,
+    AllocationStatusChoice,
 )
 
+from coldfront.core.test_helpers.factories import AllocationStatusChoiceFactory
```
[the remark this answers is not in the document]
Comment (melonistic) [reply coldfront/plugins/qumulo/tests/test_quotas.py:16]: There are other factories that seemed useful, but I did not want to introduce more them now. Something to consider.

[the remark this answers is not in the document]
Comment (harterjwustl) [reply coldfront/plugins/qumulo/tests/test_quotas.py:16]: Yeah, good to know about.  I think it is entirely reasonable to use them so as to harness the work already put into coldfront core

**Comment by melonistic** [comment]
@harterjwustl @peaches7083 

The uniqueness validator for the allocation file system path is not dependent on the allocation status. The file system path cannot be reused for new allocations if there are existing allocations with the same path in any status, including Revoked, Expired, or Deleted. I cannot create data through the UI in QA to test the data ingestion issue this ticket is trying to fix. On the other hand, we have data in production to test this issue.

**Review by harterjwustl — APPROVED** [review]
Tested locally to prove success

**Pull request merged**